cmd/buildtools/helm.go

```diff
@@ -0,0 +1,214 @@
+		chrepo, err := repo.NewChartRepository(
+			repository, getters,
+		)
+		if err != nil {
+			return fmt.Errorf("unable to create chart repo: %w", err)
+		}
+		chrepo.CachePath = h.tmpdir
+		_, err = chrepo.DownloadIndexFile()
+		if err != nil {
+			return fmt.Errorf("unable to download index file: %w", err)
+		}
+	}
+	h.repocfg = repocfg
+	return nil
+}
+
+func (h *Helm) Close() error {
+	return os.RemoveAll(h.tmpdir)
+}
+
+func (h *Helm) Latest(reponame, chart string) (string, error) {
+	logrus.Infof("finding latest version of %s/%s", reponame, chart)
+	for _, repository := range repositories.Repositories {
+		if repository.Name != reponame {
+			continue
+		}
+		chrepo, err := repo.NewChartRepository(repository, getters)
+		if err != nil {
+			return "", fmt.Errorf("unable to create chart repo: %w", err)
+		}
+		chrepo.CachePath = h.tmpdir
+		idx, err := chrepo.DownloadIndexFile()
+		if err != nil {
+			return "", fmt.Errorf("unable to download index file: %w", err)
+		}
+
+		repoidx, err := repo.LoadIndexFile(idx)
+		if err != nil {
+			return "", fmt.Errorf("unable to load index file: %w", err)
+		}
+
+		versions, ok := repoidx.Entries[chart]
+		if !ok {
+			return "", fmt.Errorf("chart %s not found", chart)
+		} else if len(versions) == 0 {
+			return "", fmt.Errorf("chart %s has no versions", chart)
+		}
+
+		var outerr error
+		sort.Slice(versions, func(i, j int) bool {
+			v1, err1 := normalizeVersion(versions[i].Version)
+			v2, err2 := normalizeVersion(versions[j].Version)
+			if err1 != nil || err2 != nil {
+				if outerr == nil {
+					outerr = fmt.Errorf("%s, %s", err1, err2)
+				}
+				return false
+			}
+			return v1.LessThan(*v2)
+		})
+
+		if outerr != nil {
+			return "", fmt.Errorf("unable to sort versions: %w", outerr)
+		}
+		return versions[len(versions)-1].Version, nil
+	}
+	return "", fmt.Errorf("repository %s not found", reponame)
+}
+
+func (h *Helm) Pull(repo, chart, version string) (string, error) {
+	if err := h.prepare(); err != nil {
+		return "", err
+	}
+
+	dl := downloader.ChartDownloader{
+		Out:              io.Discard,
+		Options:          []getter.Option{},
+		RepositoryConfig: h.repocfg,
+		RepositoryCache:  h.tmpdir,
+		Getters:          getters,
+	}
+	ref := fmt.Sprintf("%s/%s", repo, chart)
+	dst, _, err := dl.DownloadTo(ref, version, os.TempDir())
+	if err != nil {
+		return "", err
+	}
+	return dst, nil
+}
+
+func (h *Helm) RegistryAuth(server, user, pass string) error {
+	return h.regcli.Login(server, registry.LoginOptBasicAuth(user, pass))
+}
+
+func (h *Helm) Push(path, dst string) error {
+	up := uploader.ChartUploader{
+		Out:            os.Stdout,
+		Pushers:        pushers,
+		RegistryClient: h.regcli,
+	}
+	return up.UploadTo(path, dst)
+}
+
+func normalizeVersion(version string) (*semver.Version, error) {
+	parts := strings.Split(version, ".")
+	for len(parts) < 3 {
+		parts = append(parts, "0")
+	}
+	paddedVersion := strings.Join(parts, ".")
+	return semver.NewVersion(paddedVersion)
+}
```

Comment:
we should add a comment that this is only intended for sorting version strings, and not for storing/accessing them